drbd_metrics.go

```diff
@@ -65,6 +68,28 @@ func (c *drbdCollector) Collect(ch chan<- prometheus.Metric) {
 
 	log.Infoln("Collecting DRBD metrics...")
 
+	// set split brain metric
+	// by default if the custom hook is not set, the exporter will not be able to detect it
+	files, err := ioutil.ReadDir("/var/run/drbd/splitbrain")
+	if err != nil {
+		log.Warnf("Error while reading directory /var/run/drbd/splitbrain: %s", err)
+	} else {
+		for _, f := range files {
+			// check if in directory there are file of syntax we expect (nil is when there is not any)
+			match, _ := filepath.Glob("/var/run/drbd/splitbrain/drbd-split-brain-detected-*")
+			if match != nil {
+				resAndVolume := strings.Split(f.Name(), "drbd-split-brain-detected-")[1]
+				// avoid to have index out range panic error (in case the there is not resource-volume syntax)
+				if len(strings.Split(resAndVolume, "-")) == 2 {
+					resource := strings.Split(resAndVolume, "-")[0]
+					volume := strings.Split(resAndVolume, "-")[1]
+					ch <- c.makeGaugeMetric("split_brain", float64(1), resource, volume)
+					log.Warnf("Found split brain for drbd %s %s! Please remove file once you have resolved splitbrain", resource, volume)
+				}
+			}
+		}
+	}
+
```

Comment:
So far, so good! Though, there are still a few things that I'd like to see improved in the implementation:

1. I think the directory should be configurable and not hard-coded! I really don't like hard-coded stuff, and now that we have a configuration system in place we have no excuses... ;)
2. The `Collect` one has gotten quite big. I think we should refactor it like the pacemaker collector, and have the two main collection blocks separated, e.g. `c.recordResourceMetrics` and `c.recordSplitBrainMetric`.
3. Rather than nesting all these `if` blocks one into another, consider the  "fail fast, fail early" approach: that would make the most relevant part of the code stand out, instead of being hidden by multiple levels of indentation.

Reply:
Yep I agree. The thing about 3) is that I might need to

THe thing is that the drbd collector has 2 sources of inputs. ( the binary drbd

Reply:
done